op-node/rollup/derive/channel.go

```diff
@@ -28,10 +25,8 @@ type Channel struct {
 	// true if we have buffered the last frame
 	closed bool
 
-	// TODO: implement this check
 	// highestFrameNumber is the highest frame number yet seen.
-	// This must be equal to `endFrameNumber`
-	// highestFrameNumber uint16
+	highestFrameNumber uint16
```

Comment:
Could a malicious sequencer in theory overflow this value?

Reply:
No, it's a uint16 on the wire: https://github.com/ethereum-optimism/optimism/blob/d734f9d93ccbe12c2fa51d0a9fab772764493fcc/op-node/rollup/derive/frame.go#L75